go/vt/vtgate/planbuilder/operators/subquery_planning.go

```diff
@@ -43,7 +43,7 @@ func isMergeable(ctx *plancontext.PlanningContext, query sqlparser.SelectStateme
 
 	sel, ok := query.(*sqlparser.Select)
 	if !ok {
-		return false
+		return true
 	}
 
```

Comment:
below there is a check for aggregations. A union query can still have aggregations, not sure if it can be merged without checking that.